tls/hello_message.py

```diff
@@ -72,6 +91,9 @@ def from_bytes(cls, bytes):
         :return: ClientHello object.
         """
         construct = _constructs.ClientHello.parse(bytes)
+        if any(extension.type not in cls.allowed_extensions
```

Comment:
This was an excellent catch!  Should `as_bytes` also have this check, so that it's not possible to serialize a hello message that contains invalid extensions?